procrastinate/testing.py

```diff
@@ -107,21 +107,23 @@ async def defer_jobs_all(self, jobs: list[types.JobToDefer]) -> list[JobRow]:
         # We check the queueing locks upfront so that no job is inserted into
         # the queue if the constraint is violated (simulating a database
         # rollback).
-        seen_queueing_locks = set()
-        for job in jobs:
-            queueing_lock = job.queueing_lock
-            if queueing_lock is not None and (
-                any(
-                    job["queueing_lock"] == queueing_lock and job["status"] == "todo"
-                    for job in self.jobs.values()
-                )
-                or queueing_lock in seen_queueing_locks
-            ):
-                raise exceptions.UniqueViolation(
-                    constraint_name=manager.QUEUEING_LOCK_CONSTRAINT
-                )
+        current_queuing_locks = {
+            job["queueing_lock"]
+            for job in self.jobs.values()
+            if job["status"] == "todo"
+        } - {None}
 
-            seen_queueing_locks.add(queueing_lock)
+        new_queueing_locks = [
+            job.queueing_lock for job in jobs if job.queueing_lock is not None
+        ]
+        new_queueing_locks_set = set(new_queueing_locks)
+        if (
+            len(new_queueing_locks) != len(new_queueing_locks_set)
+            or current_queuing_locks & new_queueing_locks_set
```

Comment:
(ah good, you spotted and fixed the typo in my suggestion that I just saw :D 👏  Sorry about that)

Reply:
> (ah good, you spotted and fixed the typo in my suggestion that I just saw :D 👏 Sorry about that)

No, prob. It's good that occasionally I also have to switch my brain on 😉. Unfortunately, I had to rewrite it because of how the queueing lock name is now handled.